velox/exec/Operator.cpp

```diff
@@ -97,6 +97,26 @@ void Operator::maybeSetReclaimer() {
       Operator::MemoryReclaimer::create(operatorCtx_->driverCtx(), this));
 }
 
+void Operator::maybeSetQueryTraceWriter() {
+  // TODO: Make this branch unlikely.
+  if (const auto& queryTraceConfig =
+          operatorCtx_->driverCtx()->queryTraceConfig()) {
+    if (queryTraceConfig->queryNodes.count(operatorType()) != 0) {
+      const auto subDir = fmt::format(
+          "{}/{}/{}/data",
+          planNodeId(),
+          operatorCtx_->driverCtx()->pipelineId,
+          operatorCtx_->driverCtx()->driverId);
+      const auto traceDir =
+          operatorCtx_->driverCtx()->task->createTraceDirectory(subDir);
```

Comment:
We could also let createTraceDirectory take necessary inputs to create the sub directory path.